### ISSUE TITLE
404 message has a misleading pkgId when using a private registry

### ISSUE THREAD

**Issue by shayke** [npm/npm-registry-client#80]
This is a problem when using private repo host with a subpath.
Scenario (my registry is set to `http://localhost/api/npm/npm-local`):

```
npm install notfoundpkg --registry http://localhost/api/npm/npm-local
npm ERR! 404 Not Found: api
npm ERR! 404 
npm ERR! 404 'api' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.
```

Now, in `lib/request.js` I see the following snippet:

``` javascript
var w = url.parse(where).pathname.substr(1)
var name
if (!w.match(/^-/)) {
  w = w.split("/")
  name = w[w.indexOf("_rewrite") + 1]
}
```

I'm not sure why it's doing indexOf("_rewrite") but in the `npm install` case:
w="api/npm/npm-local/express"
so it should take the last part after splitting by "/".

I could post a pull request for that but I just don't know about the _rewrite thingie..

**Comment by jayphelps** [comment]
What's the word on this? Did #8257 supposedly resolve this? Still seeing it in `v3.3.4`

**Comment by cstamas** [comment]
Still here in `v3.5.3`

**Comment by trojanc** [comment]
Also still seeing this in ```v3.8.5```

**Comment by marmax** [comment]
Same here with 3.8.9 :(

**Comment by backsapce** [comment]
same here with 3.10.5 :(

**Comment by Nevraeka** [comment]
still seeing this in `v.4.6.1`

**Comment by zkat** [comment]
This is getting fixed in npm5. Wait another week :)

**Comment by basst314** [comment]
Hi @zkat, has this already been fixed by now? Are there fixes for older npm-Versions aswell? Thanks!

**Comment by lexrj** [comment]
This appears to be fixed now (``5.0.4``).

**Comment by zkat** [comment]
This should be fixed now, and if it isn't, it's definitely no longer the responsibility of this package -- pacote is what does this now. Closing.
